src/NServiceBus.Metrics.PerformanceCounters/Counters/PerformanceCountersCache.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+
+class PerformanceCountersCache : IDisposable
+{
+    public IPerformanceCounterInstance Get(CounterInstanceName counterInstanceName)
+    {
+        var counter = counterCache.GetOrAdd(counterInstanceName, CreateInstance);
+
+        return counter;
+    }
+
+    public void Dispose()
+    {
+        foreach (var counter in counterCache.Values)
+        {
+            counter.Dispose();
+        }
+
+        counterCache.Clear();
+    }
+
+    protected virtual IPerformanceCounterInstance CreateInstance(CounterInstanceName counterInstanceName)
+    {
+        PerformanceCounter counter;
+
+        if (counterInstanceName.InstanceName.Length > sbyte.MaxValue)
+        {
+            throw new Exception($"The endpoint name ('{counterInstanceName.InstanceName}') is too long (longer then {sbyte.MaxValue}) to register as a performance counter instance name. Reduce the endpoint name.");
```

Comment:
Remove the ( ) from the exception message.

Another suggestion: "Reduce the endpoint name" is not necessary.